### ISSUE TITLE
reproducible quantile regression predictions require R set.seed() before ranger() call

### ISSUE THREAD

**Issue by mkb007** [imbs-hl/ranger#643]
The documentation for quantile regression should emphasize that reproducible predictions are only possible if R's set.seed is called before training the forest using ranger().   R's sample() function is used at the end of ranger() to initialize the random.node.values of the ranger object, which are used by predict.ranger(type = "quantiles").  The potential confusion exists because there are two RNGs - one for R and another for C++.

See issue #637.

**Comment by mnwright** [comment]
Good point. I still think a refactor of the RNG would be good to improve flexibility, see e.g. here: https://github.com/imbs-hl/ranger/issues/533#issuecomment-657465666.

**Comment by mkb007** [comment]
Yes.  The encapsulated RNG idea would help reproducibility and reduce some possible wasted effort with quantile regression.

**Comment by UnixJunkie** [comment]
Great! Thanks for the fix!
